### PR TITLE
Knora-Sipi integration tests

### PR DESCRIPTION
**Pull request by benjamingeer**
This replaces the Python scripts in `webapi/_test_data/test_route` with tests in Scala under `it`.

It also refactors `ResourcesRouteV1` and `ValuesRouteV1` to get rid of `var`.

To run the Knora-Sipi integration tests, first start Sipi like this:

```
local/bin/sipi --config config/sipi.knora-config.lua
```

Then open an SBT console in `Knora/webapi` and type:

```
it:test-only *SipiV1ITSpec
```

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
Did you solve the chunked problem by giving a content length?

**Comment by benjamingeer** [comment]
> Did you solve the chunked problem by giving a content length?

No, I haven't done anything about that.

**Comment by benjamingeer** [comment]
The Sipi bug that was blocking this (https://github.com/dhlab-basel/Sipi/issues/121) was fixed in https://github.com/dhlab-basel/Sipi/pull/130.

File: webapi/src/it/scala/org/knora/webapi/e2e/v1/SipiV1ITSpec.scala [hunk 1]
```diff
@@ -175,63 +128,225 @@ class SipiV1ITSpec extends ITSpec(SipiV1ITSpec.config) with TriplestoreJsonProto
...
+                  |        "http://www.knora.org/ontology/incunabula#origname": [
+                  |            {"richtext_value": {"utf8str": "Chlaus"}}
+                  |        ],
+                  |        "http://www.knora.org/ontology/incunabula#partOf": [
+                  |            {"link_value": "http://data.knora.org/5e77e98d2603"}
+                  |        ],
+                  |        "http://www.knora.org/ontology/incunabula#seqnum": [{"int_value": 99999999}]
+                  |    },
+                  |    "file": {
+                  |        "originalFilename" : "${jsonFields("original_filename").asInstanceOf[JsString].value}",
```
Comment (tobiasschweizer) [inline webapi/src/it/scala/org/knora/webapi/e2e/v1/SipiV1ITSpec.scala]: Isn't there a way to embed a JsObject representing the file and serialise to a JSON string (calling `compactPrint` on it)?

Like in the test `change an 'incunabula:page' with parameters`.

Something like this:
```
val fileParams = JsObject(
                Map(
                    "originalFilename" -> jsonFields("original_filename"),
                    "originalMimeType" -> jsonFields("original_mimetype"),
                    "filename" -> jsonFields("filename")
                )
            )

            val knoraParams =
                s"""
                   |{
                   |    "restype_id": "http://www.knora.org/ontology/incunabula#page",
                   |    "properties": {
                   |        "http://www.knora.org/ontology/incunabula#pagenum": [
                   |            {"richtext_value": {"utf8str": "test page"}}
                   |        ],
                   |        "http://www.knora.org/ontology/incunabula#origname": [
                   |            {"richtext_value": {"utf8str": "Chlaus"}}
                   |        ],
                   |        "http://www.knora.org/ontology/incunabula#partOf": [
                   |            {"link_value": "http://data.knora.org/5e77e98d2603"}
                   |        ],
                   |        "http://www.knora.org/ontology/incunabula#seqnum": [{"int_value": 99999999}]
                   |    },
                   |    "file": ${fileParams.compactPrint},
                   |    "label": "test page",
                   |    "project_id": "http://data.knora.org/projects/77275339"
                   |}
                """.stripMargin
```

File: webapi/src/it/scala/org/knora/webapi/ITSpec.scala [hunk 2]
```diff
@@ -60,11 +62,27 @@ class ITSpec(_system: ActorSystem) extends Core with KnoraService with Suite wit
     implicit protected val ec: ExecutionContextExecutor = system.dispatcher
     implicit protected val materializer = ActorMaterializer()
 
-    def singleAwaitingRequest(request: HttpRequest, duration: Duration = 3.seconds): HttpResponse = {
+    protected def singleAwaitingRequest(request: HttpRequest, duration: Duration = 5.seconds): HttpResponse = {
```
Comment (benjamingeer) [inline webapi/src/it/scala/org/knora/webapi/ITSpec.scala]: I haven't had this problem, what duration did you use?

File: webapi/src/it/scala/org/knora/webapi/e2e/v1/SipiV1ITSpec.scala (hunk 1 above)
Comment (benjamingeer) [inline webapi/src/it/scala/org/knora/webapi/e2e/v1/SipiV1ITSpec.scala]: OK

**Comment by benjamingeer** [comment]
> Could we think about adding a delete route to Sipi to automatically remove the files created during the test? see dhlab-basel/Sipi#26

I've replied in that issue.

File: webapi/src/it/scala/org/knora/webapi/ITSpec.scala (hunk 2 above)
Comment (tobiasschweizer) [inline webapi/src/it/scala/org/knora/webapi/ITSpec.scala]: I increased it to 15 seconds and they worked fine. 

I ran the tests on my MacBookAir.

**Comment by tobiasschweizer** [comment]
I ran the tests on my faster iMac. They pass without problems with a duration of 5s.

File: webapi/src/it/scala/org/knora/webapi/e2e/v1/SipiV1ITSpec.scala (hunk 1 above)
Comment (tobiasschweizer) [inline webapi/src/it/scala/org/knora/webapi/e2e/v1/SipiV1ITSpec.scala]: You can do this rather stylistic improvement if you think it makes sense and then merge.

**Pull request merged**